### DISCUSSION TITLE
@fluid-experimental/data-objects compiles to esnext module only

### DISCUSSION THREAD

**Discussion by SampoSyrjanen** [microsoft/FluidFramework#19572] [General]
Why is https://github.com/microsoft/FluidFramework/blob/main/experimental/framework/data-objects/tsconfig.json compiling a "esnext" module?

**Comment by jason-ha** [comment]
@SampoSyrjanen, that is a fair question. The ESM/CJS story has not been great for FF. A major effort is going on right now to straighten things up especially around Node16 resolution and will clarify CJS/ESM support. Since experimental data-objects has been ESM only, it probably remains ESM during this effort. Long term the idea is just to support ESM.

**Reply by SampoSyrjanen** [reply]
Ok, we make use of the CJS for Node as it is now compiled in other FF packages. A workaround now is to copy the source of the experimental package to our own compilation to make it work with CJS. But yeah, long term goal is probably to make everything ESM.
